File: GGAndroid_2.0/GGPay/pay-wxpay/src/main/java/io/ganguo/pay/wxpay/WXPayMethod.java
package io.ganguo.pay.wxpay;

import android.app.Activity;

import io.ganguo.pay.core.GGPay;
import io.ganguo.pay.core.IPayService;
import io.ganguo.pay.core.PayCallback;

/**
 * Created by Roger on 07/07/2017.
 */

public class WXPayMethod implements IWXPayMethod {

    public WXPayMethod(String wechatAppId) {
        if (wechatAppId == null || wechatAppId.equals("")) {
            throw new IllegalArgumentException("No wechat app id registered in GGPay");
        }
        WXConstants.setWeChatAppId(wechatAppId);
    }

    @Override
    public void pay(Activity activity, PayCallback payCallback, WXPayData wxPayData) {
        GGPay.newService(new WXPayServiceProvider(activity, payCallback, wxPayData)).pay();
    }

    @Override
    public IPayService asPayService(Activity activity, PayCallback payCallback, WXPayData wxPayData) {
        return GGPay.newService(new WXPayServiceProvider(activity, payCallback, wxPayData));
    }
}

File: GGAndroid_4.0/core/service/service-pay/pay-wx/src/main/kotlin/io/ganguo/pay/wxpay/WXPayMethod.kt
package io.ganguo.pay.wxpay

import android.app.Activity
import io.ganguo.pay.wxpay.entity.WXPayEntity

/**
 * Created by Roger on 07/07/2017.
 * 微信支付
 * 微信支付结果回调处理在[WXPayActivity]
 */
class WXPayMethod(weChatAppId: String) : AWXPayMethod() {

    init {
        WXConstants.WX_APP_ID = weChatAppId
        WXConstants.WX_SDK_IS_INIT = WXConstants.WX_APP_ID.isNotEmpty()
    }


    /**
     * WXPayService 创建方法
     * @param p 支付订单请求数据
     * @return
     */
    override fun newPayProvider(activity: Activity, param: WXPayEntity?): WXPayProvider {
        return WXPayProvider(activity, param!!)
    }
}
